myia/utils/get_fields.py

```diff
@@ -0,0 +1,22 @@
+"""Miscellaneous utilities."""
+
+from dataclasses import is_dataclass
+
+from .misc import dataclass_fields
+from .overload import overload
+
+
+@overload
+def get_fields(instance: object):
+    """Returns fields of an instance."""
+    if is_dataclass(instance):
+        return dataclass_fields(instance).items()
```

Comment:
```suggestion
        return dataclass_fields(instance)
```

The overload for Module returns a dictionary, and _explore expects a dictionary.